draft-ietf-core-yang-cbor.md

```diff
@@ -152,7 +152,10 @@ In order to minimize the size of the encoded data, the proposed
 mapping avoids any unnecessary meta-information beyond that directly
 provided by the CBOR basic generic data model ({{Section 2 of RFC8949}}). For instance, CBOR tags are used solely in the case of a SID not encoded as delta, anyxml schema nodes, or the union datatype, to distinguish explicitly the use of different YANG datatypes encoded using the same CBOR major type.
 
-Unless specified otherwise by the protocol or mechanism implementing this specification, the indefinite lengths encoding as defined in {{Section 3.2 of RFC8949}} SHALL be supported by CBOR decoders.
+Unless specified otherwise by the protocol or mechanism implementing this specification, the indefinite length encoding as defined in {{Section 3.2 of RFC8949}} SHALL be supported by the CBOR decoders employed with YANG-CBOR.
+(This enables an implementation to begin emitting an array or map
+before the number of entries in that structure is known, possibly also
+avoiding excessive locking or race conditions.)
```

Comment:
Should this text establish any advice about limits on size for transmitted maps?
Or advice about providing advice?
Indefinite maps that yang-data/on-disk have a natural upper limit on how big the map can be: the size of the file.
I think that we constrained-voucher use case would be happy with indefinite length encoding, but if I have only 4k to deal with a response, I would rather know up-front, if I'm gonna be able to cope with that 10000 element map?

Reply:
Well, we either can make things easy for the generator by enabling indefinite length, or make it easy for the consumer by making size information available beforehand, which has all the onus on the generator alluded to in the parenthesis.
Or we could explain when which decision is right, but that will be hard to get right.

Reply:
See 7a5a6fc for an attempt to clarify this trade-off.

Reply:
Indefinite length is needed for YANG Push because the specific data nodes sent to the client
are usually not known in advance, and not buffered in memory by the server.  Nodes can be omitted
based on complex XPath filters and other implementation criteria.  The sender must be 100% correct about
the sizes or the message is corrupt and will be dropped by the receiver.  Therefore YANG Push servers will probably
use indefinite form for every container and list.